app/controllers/staff/submissions_controller.rb

```diff
@@ -65,4 +58,17 @@ def activate_submission
       @submission.create_activity key: "submission.activate", owner: current_user
     end
   end
+
+  def send_csv
+    response.headers["Content-Type"] = "text/event-stream"
```

Comment:
I like the idea of streaming this row-by-row, that's a nice improvement and I didn't know Rails directly supported streaming now. I'm not sure about the MIME type here as `text/event-stream` is typically used specifically for the Server-Sent Events protocol. Is there a reason to use it here rather than `text/csv`? Does it cause the browser to handle the download differently?

Reply:
Do you know what, I don't know - all the examples I found of this implementation used `event-stream` so I just went with it! But it turns out that `text/csv` works as well so I'll set it back to that.

Reply:
I think the controller also needs `include ActionController::Live` added to make this work

(Source: https://api.rubyonrails.org/classes/ActionController/Live.html)

Reply:
Oops. Added.

Reply:
We've talked about this and determined that although that Rails API looks like it's specifically designed for Server-Sent Events, it does allow streaming arbitrary data and does not prefix each line with `data: `, so this implementation looks good 👍